modin/numpy/arr.py

```diff
@@ -1079,7 +1132,9 @@ def append(self, values, axis=None):
             raise ValueError(
                 f"all the input arrays must have same number of dimensions, but the array at index 0 has 1 dimension(s) and the array at index 1 has {values._ndim} dimension(s)"
             )
-        if self.shape[axis ^ 1] != values.shape[axis ^ 1]:
+        if (axis ^ 1 < len(values.shape)) and self.shape[axis ^ 1] != values.shape[
```

Comment:
Wouldn't it be more obvious to check dimensions? (instead of `(axis ^ 1 < len(values.shape))`)

Reply:
I assume you're suggesting replacing `(axis ^ 1 < len(values.shape))` with `values._ndim == 2`? The issue with that is we should still take this branch if `ndim == 1 and axis == 1`; the first clause is just to avoid an IndexError on `values.shape` in the second clause.

Reply:
I mean something like this: `(axis ^ 1 < values._ndim)`.

Reply:
By the way, as far as I know, numpy supports more than 2 axes. Do we have a check that `axis<2`? In the case when `axis>=2`, the `^` operation will not work as intended.

Reply:
The case where `axis>=2` on 2D/1D arrays is caught within the internal `concat` method call, though the error message does end up different. In general, we do not currently support arrays with dimension higher than 2, which is validated in the [array constructor](https://github.com/modin-project/modin/blob/16d938099257a6317b72b53762d687a651531cd9/modin/numpy/arr.py#L176).

Reply:
ok, @noloerino thanks for the clarification!